ForumProject/subscriptions/views.py

```diff
@@ -1,3 +1,199 @@
+        if user_investor:
+            # Check if subscription already exists for this investor and startup
+            existing_subscription = SubscribeInvestorStartup.objects.filter(
+                investor=user_investor.investor,
+                startup_id=startup_id
+            ).exists()
+            if existing_subscription:
+                return Response({'error': 'Subscription already exists'}, status=status.HTTP_400_BAD_REQUEST)
+
+            serializer = self.get_serializer(data={'startup': startup_id})
+            serializer.is_valid(raise_exception=True)
+            serializer.save(investor=user_investor.investor)
+            return Response({'message': 'Subscription added successfully.'}, status=status.HTTP_201_CREATED)
+        else:
+            return Response({'error': 'User is not associated with an investor'}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class ListMySubscription(viewsets.ReadOnlyModelViewSet):
+    """
+    View for listing user's subscriptions.
+
+    Serializer class: SubscribeInvestorStartupSerializer
+    Filter backends: DjangoFilterBackend, SearchFilter
+    Filterset class: MySubscriptionFilter
+    Search fields: startup__startup_name, startup__startup_industry, startup__startup_country
+    """
+    serializer_class = SubscribeInvestorStartupSerializer
+    filter_backends = [DjangoFilterBackend, filters.SearchFilter]
+    filterset_class = MySubscriptionFilter
+    search_fields = ['^startup__startup_name', '^startup__startup_industry', '=startup__startup_country']
+    # permission_classes = [IsInvestorCompanySelected]
+
+    def get_queryset(self):
+        """
+        Get the queryset for user's subscriptions.
+
+        Returns:
+            Queryset: Subscriptions related to the user's investor profile.
+        """
+        user = self.request.user
+        try:
+            user_investor = UserInvestor.objects.get(customuser=user)
+            investor_id = user_investor.investor_id
+            queryset = SubscribeInvestorStartup.objects.filter(investor_id=investor_id)
+            return queryset
+        except UserInvestor.DoesNotExist:
+            return SubscribeInvestorStartup.objects.none()
+
+    def list(self, request, *args, **kwargs):
+        """
+        List user's subscriptions with startup details.
+
+        Returns:
+            Response: List of subscriptions with startup details.
+        """
+        queryset = self.filter_queryset(self.get_queryset())
+        serializer = self.get_serializer(queryset, many=True)
+        response_data = []
+        for subscription in serializer.data:
+            startup_id = subscription['startup']
+            startup = Startup.objects.get(id=startup_id)
+            startup_serializer = StartupSerializer(startup)
+            subscription['startup'] = startup_serializer.data
+            response_data.append(subscription)
+        return Response(response_data)
```

Comment:
Instead of fetching `Startup` objects in the loop, consider pre-fetching related models to reduce queries

Reply:
Done, thanks!